src/main/java/ru/hogwarts/school/controller/StudentController.java

```diff
@@ -0,0 +1,57 @@
+package ru.hogwarts.school.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ru.hogwarts.school.model.Student;
+import ru.hogwarts.school.service.StudentService;
+
+import java.util.Collection;
+import java.util.List;
+
+@RestController
+@RequestMapping("/student")
+public class StudentController {
+    @Autowired
+    private final StudentService studentService;
+
+    public StudentController(StudentService studentService) {
+        this.studentService = studentService;
+    }
+
+    @GetMapping("{id}")
+    public Student getStudentInfo(@PathVariable Long id) {
+        return studentService.findStudent(id);
+    }
+
+    @PostMapping
+    public Student createStudent(@RequestBody Student student) {
+        return studentService.createStudent(student);
+    }
+
+    @PutMapping
+    public ResponseEntity<Student> editStudent(@RequestBody Student student) {
+        Student foundStudent = studentService.editStudent(student);
+        if (foundStudent == null) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+        }
+        return ResponseEntity.ok(foundStudent);
+    }
+
+    @DeleteMapping("{id}")
+    public ResponseEntity<Student> deleteStudent(@PathVariable("id") Long id) {
+        studentService.deleteStudent(id);
+        return ResponseEntity.ok().build();
+    }
+
+    @GetMapping
+    public List<Student> findByAge(@RequestParam int age) {
+        return studentService.findByAge(age);
+    }
+
+    @GetMapping("/all")
```

Comment:
```suggestion
    @GetMapping
```
никаких слов быть не должно